x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ldap/SearchGroupsResolverInMemoryTests.java

```diff
@@ -77,6 +79,10 @@ public void testSearchTimeoutIsFailure() throws Exception {
             assertThat(((LDAPException) cause).getResultCode(), is(ResultCode.TIMEOUT));
         } finally {
             ldapServers[0].setProcessingDelayMillis(0);
+
+            // Wait for the async search to complete or timeout.
+            // Without this, we might trigger an orphaned thread - see https://github.com/pingidentity/ldapsdk/issues/120
+            assertBusy(() -> assertThat(connection.getActiveOperationCount(), lessThanOrEqualTo(0)), 1000, TimeUnit.MILLISECONDS);
```

Comment:
Nit: Would it be useful to increase the timeout a bit more, like 2000 ms? The server delays processing for 500ms, this means we have only one chance (at 512ms) to get a zero back from the `getActiveOperationCount()` because `assertBusy` fires with a power of two time interval. If there is any delay on the LDAP server side, the check at 512 ms could fail?

Reply:
```suggestion
            assertBusy(() -> assertThat(connection.getActiveOperationCount(), lessThanOrEqualTo(0)), 2000, TimeUnit.MILLISECONDS);
```